### DISCUSSION TITLE
Config stored on RX on each quad

### DISCUSSION THREAD

**Discussion by rodrigosclosa** [ExpressLRS/ExpressLRS#2047] [Ideas]
Hey guys, how are you doing?

I've been a user and promoter of the project since when it was just an idea, nothing was ready and we had to assemble our hardware...

But anyway, I updated all my quads recently to version 3.1.2 and an idea came to me...

Would it be possible to store the ELRS configuration in the Receiver of each drone, instead of having to change it every time we change drones? I'll give you an example: I have a cinematic filming drone, which I use with a packet rate setting of 150Hz, telemetry in 1:64... I have another Freestyle drone that I fly with a packet rate of 500Hz and telemetry 1:16. ..
This configuration of each model, store in the RX itself, and when the bind is done, send this config to the TX and automatically set the configuration...

Another point is that in BF 4.3, using the presets, we have to choose the packet rate update rate because of the filters and other things... It would help a lot if the RX saved the config and sent it to the TX when the bind was done ...

What do you think?

**Comment by Dropsleutel** [comment]
So basically if we use Modelmatch all elrs setting should be stored for each ELRS setting ? Isit not yet the case ?
At Tx you just select the Quad you want to fly ?

From my side, to setup channels, endpoinds low high thresholds asign switches on both sides is annoying..  Would be nice if we can assign/define in Edge Tx our controls Arms switches our stick axis our endpoints with labels etc in Betaflight nothing to set just select ELRS ? (Possibly a  library in EdgeTx, select Betaflight, library, Ardupilot library with all available switch/stick labels options controls options etc, in Edge Tx you than pick from the library assign/define/label any available function, to switches, then it matches the Fc and no more need to setup both sides Fc ?

**Reply by rodrigosclosa** [reply]
I don't know if modelmatch can do this.. Sincerelly, i don't use it... I'll research more about it...

**Comment by pkendall64** [comment]
Basically what you are asking for is model configurations (with or without model match), which we already have.
On the radio you just setup another model using a different receiver number. The receiver number specifies an index into the transmitter configurations, of which you can have up to 64 of them.
See https://www.expresslrs.org/3.0/software/model-config-match/?h=model#model-configuration for more details.

**Reply by rodrigosclosa** [reply]
But this is the point... We need to config another model on Radio and assign a number to it under External Module configurations...
But I use exactly the same switches, channels and mixes configuration all over my quads... So, I have only one model called "ELRS", that I use for all my quads... Using modelmatch, I need to duplicate this model for all my 8 quads... That's what I don't want to do...

My suggestion here, is to store these ELRS config into RECEIVER directly, without use of modelmatch... For example, as DJI does with Air Units, it stores the channel, bitrate, etc, all on Air Unit side as we change on googles... As it boots up and got binded, the googles are configured with these stored parameter from this particullary Air Unit...

**Reply by pkendall64** [reply]
I wouldn't do it for all the quads. I'd do by flight style. i.e. 
- Freestyle: 500Hz Std tlm, Receiver number 1
- Long Range: 100Hz 1:8 tlm Receiver number 2
And don't turn model match on, then each quad can be used in either model configuration.

Model match on just means that the receiver will only operate if the model numbers on the TX and RX match.
Model match off means the receiver number is just used to select communication parameters.

**Reply by rodrigosclosa** [reply]
Yes, I understand it... 

But wouldn't be nice if we can store these config's into RX and be able to take it to another module or radio, simple binding it.. For example, for a friend or a client that we configured all the tunnings on the quad with filter's for that packet rate, etc... Just wondering about it...

**Comment by mogplus88** [comment]
I think this is a good idea. I fly fixed wing planes, and have discussed this concept with a few people. The theory is that, if you have three models you want to fly, you switch on your transmitter and model A, and... go fly. Next flight you switch on your transmitter and model B and... go fly. Next flight you switch on your transmitter and model C and... go fly. The idea is that the setup for each model is stored in the receiver to match what the transmitter is sending. So there is nothing in the transmitter to select. There is no information about any particular model in the transmitter at all. It always sends the same thing regardless of what model is being flown. That's the way we used to do it back in the 70s before our transmitters had computers in them. ;-)
With telemetry you could get the receiver to transmit a model name back to the tranny that it could display on a screen maybe, but that would just be a nice to have.
